pkg/ibmcsidriver/node.go

```diff
@@ -473,6 +478,36 @@ func (csiNS *CSINodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeE
 	if len(volumePath) == 0 {
 		return nil, commonError.GetCSIError(ctxLogger, commonError.EmptyVolumePath, requestID, nil)
 	}
+
+	volumeCapability := req.GetVolumeCapability()
+	// VolumeCapability is optional, if specified, use that as source of truth.
+	if volumeCapability != nil {
+		volumeCapabilities := []*csi.VolumeCapability{volumeCapability}
+		if !areVolumeCapabilitiesSupported(volumeCapabilities, csiNS.Driver.vcap) {
+			return nil, commonError.GetCSIError(ctxLogger, commonError.VolumeCapabilitiesNotSupported, requestID, nil)
+		}
+		if blk := volumeCapability.GetBlock(); blk != nil {
+			// Noop for block NodeExpandVolume.
+			klog.V(4).InfoS("NodeExpandVolume: called. Since it is a block device, ignoring...", "volumeID", volumeID, "volumePath", volumePath)
+			return &csi.NodeExpandVolumeResponse{}, nil
+		}
+	} else {
+		// VolumeCapability is nil, check if volumePath points to a block device.
+		isBlock, err := csiNS.Stats.IsBlockDevice(volumePath)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to determine if volumePath [%v] is a block device: %v", volumePath, err)
+		}
+		if isBlock {
+			// Skip resizing for Block NodeExpandVolume
+			capacity, err := csiNS.Stats.DeviceInfo(volumePath)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get block capacity on path %s: %v", volumePath, err)
+			}
+			klog.V(4).InfoS("NodeExpandVolume: called, since given volumePath is a block device, ignoring...", "volumeID", volumeID, "volumePath", volumePath)
+			return &csi.NodeExpandVolumeResponse{CapacityBytes: capacity}, nil
```

Comment:
Why this NodeExpandVolumeResponse has CapacityBytes and the other for `// Noop for block NodeExpandVolume` does not? Both are block volumes and IMO should have the same response.

Reply:
This could be written better, see the new version.